src/search/airportsearch.cpp

```diff
@@ -744,14 +750,166 @@ void AirportSearch::randomFlightplanClicked()
+      double dist;
+
+      do
+      {
+        destinationSuccess = false;
+        float lonX, latY;
+        do
+        {
+          if(triedIndexDestination.count() == countResult)
+            goto destinationsEnd;                                                 // all destinations have been depleted, try a different departure
+          if(triedIndexDestination.count() < randomLimit)                         // the "if" is inside the "while" for the destination because the indices get used up during "picking"
+          {
+            do
+            {
+              indexDestination = QRandomGenerator::global()->bounded(countResult);
+            }
+            while(triedIndexDestination.contains(indexDestination));
+          }
+          else
+          {
+            indexDestination = QRandomGenerator::global()->bounded(countResult);
+            while(triedIndexDestination.contains(indexDestination))
+            {
+              indexDestination = (indexDestination + 1) % countResult;
+            }
+          }
+          triedIndexDestination.insert(indexDestination, true);
+          lonX = data[indexDestination].second.getLonX();
+          latY = data[indexDestination].second.getLatY();
+        }
+        while(lonX == atools::geo::Pos::INVALID_VALUE || latY == atools::geo::Pos::INVALID_VALUE);
+        const double lon2 = lonX * degToRad;
+        const double lat2 = latY * degToRad;
+        dist = R_earth * qAcos(qSin(lat1) * qSin(lat2) + qCos(lat1) * qCos(lat2) * qCos(lon2 - lon1));     // http://www.movable-type.co.uk/scripts/latlong.html Spherical Law of Cosines
+        destinationSuccess = true;
+      }
+      while(dist < distanceMin || dist > distanceMax);
+destinationsEnd:
+      if(destinationSuccess)                                                      // the last triedIndexDestination might be taken but it might have passed the last condition
+      {
+        noSuccess = false;
+        continue;
+      }
+    }
+    while(noSuccess);
+allDeparturesTried:
+    if(departureSuccess)                                                          // the last triedIndexDeparture might be taken but it might have passed the last condition, destinationSuccess must be true if this "if" is reached
+    {
+      qDebug() << Q_FUNC_INFO << "random flight, index departure: " << indexDeparture
+               << ", random flight, index destination: " << indexDestination;
+
+      map::MapAirport airportDeparture;
+      map::MapAirport airportDestination;
+
+      AirportQuery *airportQuery = NavApp::getAirportQuerySim();
+
+      airportQuery->getAirportById(airportDeparture, data[indexDeparture].first);
+      airportQuery->getAirportById(airportDestination, data[indexDestination].first);
+
+      NavApp::getMainWindow()->routeNewFromAirports(airportDeparture, airportDestination);
+    }
+    else
+    {
+      QMessageBox msgBox;
+      msgBox.setText(tr("No airports found in the search result satisfying the criteria."));
```

Comment:
This QMessageBox is ok while the next one ...

Reply:
I left it :)